### PR TITLE
[DO NOT MERGE] A more or less complete script for checking the format of cpp files;

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
This pull-request is for testing my clang-format checker.

### CODE REVIEW

**Comment by therearesomewhocallmetim** [comment]
FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

File: tools/python/format_checker.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    format_files(files_to_format)
+
+    diff = git_diff()
+
+    if diff:
+        print("We have a diff, so the check has failed")
+        logging.getLogger().info(diff)
+        exit(2)
+
+
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: Too many blank lines.

File: tools/python/format_checker.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        files_to_format = " ".join(files)
+        clang_format(files_to_format)
+
+
+def git_diff():
+    return git("diff")
+
+
+def main():
+    print("I am main")
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: Do we really need it?

File: tools/python/format_checker.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    files_to_format = files_in_diff()
+    format_files(files_to_format)
+
+    diff = git_diff()
+
+    if diff:
+        print("We have a diff, so the check has failed")
+        logging.getLogger().info(diff)
+        exit(2)
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: Why not -1? Why 2?

File: tools/python/format_checker.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    return filter(filter_files, diff_files)
+
+
+def filter_files(file_name):
+    filtered_extensions = filter(lambda s: file_name.endswith(s), FILE_EXTENSIONS)
+    if not filtered_extensions:
+        return False
+
+    filtered_folders = filter(lambda s: file_name.startswith(s), EXCLUDE_FOLDERS)
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: Please, separate file_name to parts and check that the first part is a directory and not in exclusion list.

File: tools/python/format_checker.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+from integration_tests_runner import setup_jenkins_console_logger
+
+
+FILE_EXTENSIONS = [".cpp", ".hpp"]
+EXCLUDE_FOLDERS = ["base"]
+
+setup_jenkins_console_logger()
+
+def git(command):
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: s/command/args/

File: tools/python/format_checker.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    return exec_shell("git", command).strip()
+
+def clang_format(source_file):
+    return exec_shell("clang-format", "-i {0}".format(source_file))
+
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: Two blank lines between methods, obey PEP-8!

File: tools/python/format_checker.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+
+
+
+
+
+if __name__ == '__main__':
+    main()
+
```
Comment (ygorshenin) [inline tools/python/format_checker.py]: No need in trailing blank lines.

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

**Pull request closed without merging**